docs(cn): translate docs/faq-ajax.md into Chinese (#157)
reactjs/zh-hans.reactjs.org: content/docs/faq-ajax.md

category: FAQ
---

> ### 我怎样在React中发起AJAX请求{#how-can-i-make-an-ajax-call}

Reviewer: ```suggestion
### 我怎样在 React 中发起 AJAX 请求{#how-can-i-make-an-ajax-call}
```

> 在React开发中，你能使用任何流行的AJAX库，比如社区比较流行的[Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/),或者是浏览器内置的[window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。

Author: 多谢指正

> 在React开发中，你能使用任何你喜欢的 AJAX 库，比如社区比较流行的[Axios](https://github.com/axios/axios)，[jQuery AJAX](https://api.jquery.com/jQuery.ajax/)，或者是浏览器内置的[window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。

Reviewer: ```suggestion
在React开发中，你能使用任何你喜欢的 AJAX 库，比如社区比较流行的 [Axios](https://github.com/axios/axios)，[jQuery AJAX](https://api.jquery.com/jQuery.ajax/)，或者是浏览器内置的 [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。
```

Author: 已经更正，如果还有任何lint问题，请一下全部提出哈，这样我也方便一次全部进行修复

Reviewer: 好的，不好意思啊，我之前没有看出来。。🤣

Author: 哈哈哈 没事没事，没有blame你哈。 非常感谢你这么仔细的review☺️

> 在React开发中，你能使用任何你喜欢的 AJAX 库，比如社区比较流行的[Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/),或者是浏览器内置的[window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。

Reviewer: ```suggestion
在React开发中，你能使用任何你喜欢的 AJAX 库，比如社区比较流行的[Axios](https://github.com/axios/axios), [jQuery AJAX](https://api.jquery.com/jQuery.ajax/)，或者是浏览器内置的[window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。
```

Reviewer: 辛苦了

Author: 已经修复

> 在React开发中，你能使用任何你喜欢的 AJAX 库，比如社区比较流行的 [Axios](https://github.com/axios/axios)，[jQuery AJAX](https://api.jquery.com/jQuery.ajax/)，或者是浏览器内置的 [window.fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)。

Reviewer: 这个上次指出来过啦。

> ### 我应该在 React组件 的哪个生命周期函数中发起 AJAX 请求？{#where-in-the-component-lifecycle-should-i-make-an-ajax-call}

Reviewer: ```suggestion
### 我应该在 React组件的哪个生命周期函数中发起 AJAX 请求？{#where-in-the-component-lifecycle-should-i-make-an-ajax-call}
```

> 我们推荐你在[`componentDidMount`](/docs/react-component.html#mounting)这个生命周期函数中发起AJAX请求。这样做你可以很方便的使用AJAX请求返回回来数据，通过`setState`来更新组件。

Reviewer 2: ```suggestion
我们推荐你在[`componentDidMount`](/docs/react-component.html#mounting)这个生命周期函数中发起AJAX请求。这样做你可以拿到AJAX请求返回的数据并通过`setState`来更新组件。
```

Author: 多谢指正

> 我们推荐你在 [`componentDidMount`](/docs/react-component.html#mounting)这个生命周期函数中发起 AJAX 请求。这样做你可以拿到 AJAX 请求返回的数据并通过 `setState` 来更新组件。

Reviewer: ```suggestion
我们推荐你在 [`componentDidMount`](/docs/react-component.html#mounting) 这个生命周期函数中发起 AJAX 请求。这样做你可以拿到 AJAX 请求返回的数据并通过 `setState` 来更新组件。
```

> ### 用例：使用 AJAX 请求结果去改变 local state {#example-using-ajax-results-to-set-local-state}

Reviewer: ```suggestion
### 示例：使用 AJAX 请求结果去改变 local state {#example-using-ajax-results-to-set-local-state}
```

Reviewer: ref: https://github.com/reactjs/zh-hans.reactjs.org/issues/2

> 下面这个组件演示了如何在`componentDidMount`中发起AJAX call去更新组件的state。

Reviewer 3: call 还是要翻译的吧

Author: Fixed

> 示例 API 返回如下的 JSON 格式。

Reviewer: ```suggestion
示例 API 返回如下的 JSON 对象。
```

> 示例 API 返回如下的 JSON 对象。

Reviewer 4: ```suggestion
示例 API 返回如下的 JSON 对象：
```

            items: result.items
          });
        },

>         // 注意：在这里需要通过如下方式来处理错误而不是使用 catch() 去捕捉错误

Reviewer: ```suggestion
        // 注意：在这里需要在此处处理错误而不是使用 catch() 去捕捉错误
```

>         // 因为使用catch去捕捉错误会掩盖掉组件本身可能产生的 bug

Reviewer: 保持行数统一，最好也拆成三行

>         // 因为使用catch去捕捉错误会掩盖掉组件本身可能产生的 bug

Reviewer: ```suggestion
        // 因为使用 catch 去捕获异常会掩盖掉组件本身可能产生的 bug
```

            items: result.items
          });
        },

>         // 注意：在这里需要在此处处理错误

Reviewer 4: ```suggestion
        // 注意：需要在此处处理错误
```

>         // 而不是使用 catch() 去捕捉错误

Reviewer 4: ```suggestion
        // 而不是使用 catch() 去捕获错误
```
